core/shared/src/main/scala/scalaz/zio/package.scala

```diff
@@ -38,4 +38,9 @@ package object zio extends EitherCompat {
   val JustExceptions: PartialFunction[Throwable, Exception] = {
     case e: Exception => e
   }
+
+  object <*> {
+    def unapply[A, B](ab: (A, B)): Some[(A, B)] =
+      Some((ab._1, ab._2))
```

Comment:
just curious - why not Some(ab)? Seems like a redundant allocation

Reply:
It's the way pattern matching works, unfortunately.

Reply:
WDYM? I meant there is no need to construct a new tuple if you already have one.

```
@ object <*> {
    def unapply[A, B](x: (A, B)): Option[(A, B)] = Some(x)
  }
defined object <*>

@ (1,2) match {
    case x <*> y => x + ", " + y
    case (x, y) => s"!$x !$y"
    case z => s"@$z@"
  }
res1: String = "1, 2"
```

Reply:
@tkroman Sorry, thought you meant something else! You're right of course. 😄